### PR TITLE
added new faq section and fixed a few minor issues in the docs

### PR DESCRIPTION
**Pull request by kaikreuzer**
Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

File: docs/sources/development/faq.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  1. _Do I have to create XML files in `ESH-INF/thing` for all devices or is there any other option?_
+
+    No, the XML files are only one way to describe your devices. Alternatively, you can implement your own [ThingTypeProvider](https://github.com/eclipse/smarthome/blob/master/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingTypeProvider.java), through which you can provide thing descriptions in a programmatic way.
+
+  1. _For my system XY, there are so many different variants of devices. Do I really need to define a thing type for every single one of them?_
+
+    Thing types are important if you have no chance to request any structural information about the devices from your system and if you need users to manually chose a thing to add or configure (i.e. there is also no automatic discovery). The thing types that you provide will be the list the user can choose from. If your system supports auto-discovery discovery and you can also dynamically construct things (and their channels) from structural information that you can access during runtime, there is in theory no need to provide any thing type description at all.
```
Comment (dnobel) [inline docs/sources/development/faq.md]: I would not say that there is "no need to provide any thing type description at all", if all things can be discovered and types can created dynamically. 

A static description of thing types has the advantage that the user knows which kind of devices are supported, no matter if he has a device at home or not. Especially for creating dynamic documentation and a list of all supported devices in Eclipse SmartHome and openHAB the static definitions are quite helpful. So for me static descriptions should always be preferred if possible. For sure, if not all devices are known at the development time, there is no other way to do it. But I think we should give the advice to describe devices with XML and only if it is not possible, the binding can provide the thing types dynamically.

Comment (kaikreuzer) [inline docs/sources/development/faq.md]: That's why I also have "in theory" in the very same sentence :-). But you are right, we should very clearly advice to provide description whenever possible. I'll update the text accordingly.

**Comment by dnobel** [comment]
Finished Review, just one comment.

**Comment by kaikreuzer** [comment]
updated the text, as well for the previous question.

**Pull request merged**